nntrainer/layers/layer_node.cpp

```diff
@@ -373,7 +374,14 @@ void LayerNode::setInputLayers(const std::vector<std::string> &layers) {
   input_layers = std::vector<props::InputLayer>(layers.begin(), layers.end());
 }
 
-bool LayerNode::hasInputShapeProperty() const { return !input_shapes.empty(); }
+bool LayerNode::hasInputShapeProperty() const {
+  auto &input_shapes =
+    std::get<std::vector<props::InputShape>>(*layer_node_props);
+
+  return !input_shapes.empty() &&
+         std::all_of(input_shapes.begin(), input_shapes.end(),
+                     [](const auto &input) { return !input.empty(); });
```

Comment:
is this possible? this means either setting shape of 0 is possible, or we are resizing out input shape without filling it.

Reply:
This is 99.99% impossible, but as this function is not in critical path, I made it to cover some impossible scenario :p

Reply:
cool.